x-pack/legacy/plugins/siem/public/components/matrix_histogram/utils.ts

```diff
@@ -92,7 +88,7 @@ export const getCustomChartData = (
   if (mapping)
     return map((item: ChartSeriesData) => {
       const mapItem = get(item.key, mapping);
-      return { ...item, color: mapItem.color };
+      return { ...item, color: mapItem ? mapItem.color : undefined };
```

Comment:
Consider replacing the ternary here with [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html), because the `?` operator below will stop executing and return `undefined` if `mapItem`  is `null` or `undefined`:

```
return { ...item, color: mapItem?.color };
```